### PR TITLE
Use the Makefile to build and push in GH actions

### PR DESCRIPTION
**Pull request by rgolangh**
Use the Makefile to build and push in GH actions

With this change we make sure all the build and push goes through the
make. A failure in CI would likely indicate that developers fail to
build images, and vice versa, given CI and local users use the same
buidlah version - currently github runner ubuntu 22 uses buildah 1.23.1
which is bit dated.

The remaining bits that don't use the makefile is the generation of the
PR to the CD repo. The current github actions have specific code to push
stuff for workflows in helm format, and the makefile invokes a script
that does it for kustomize format. We should unit the two behaviours to
the scripts can handle both, or just make all workflows use a single
format, and then use that from the action.

There should be another PR sent to the orchestrator-helm-chart to cleanup all the tekton pipeline custom build args that with that PR will be 
defaults in the Dockerfile - https://github.com/parodos-dev/orchestrator-helm-chart/pull/221

### CODE REVIEW

**Review by masayag — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -90,13 +90,9 @@ ENABLE_PERSISTENCE ?= false
 
 # extra extensions needed for persistence at build time.
 # The extentions listed below are included in the cache in image quay.io/kiegroup/kogito-swf-builder:9.99.1.CR1 or available from maven central repository
-QUARKUS_EXTENSIONS=org.kie.kogito:kogito-addons-quarkus-jobs-knative-eventing:9.99.1.redhat-00003,org.kie.kogito:kogito-addons-quarkus-persistence-jdbc:9.99.1.redhat-00003,org.kie.kogito:kogito-addons-persistence-jdbc:9.99.1.redhat-00003,io.quarkus:quarkus-jdbc-postgresql:3.2.9.Final,io.quarkus:quarkus-agroal:3.2.9.Final,org.kie:kie-addons-quarkus-monitoring-prometheus:999-SNAPSHOT,org.kie:kie-addons-quarkus-monitoring-sonataflow:999-SNAPSHOT
 QUARKUS_DEV_EXTENSIONS=""
```
Comment (gabriel-farache) [inline Makefile:93]: if you remove the `prod` value, please also remove the `dev` one

Comment (gabriel-farache) [inline Makefile:93]: if you remove the prod value, please also remove the dev one

**Review by gabriel-farache — CHANGES_REQUESTED** [review]
1 little clean up and it's all good

BTW, the PR title does not reflects what it actually brings (uniformisation of the image and manifests generation and removal of the auto PR)

**Comment by rgolangh** [comment]
Sorry for the misleading pr title, I also noticed that, and changed it.
I'll send another PR with the actual disabling of the send and leave this
PR as a pure refactoring one.

I'll remove the DEV related args as well, and for future use I think that
we can expose a second arg to extend the arg if we want.
something like:
ARG EXTRA_QUARKUS_EXTENSIONS
ARG QUARKUS_EXTENSIONS="defaults"
ARG QUARKUS_EXTENSIONS="${QUARKUS_EXTENSIONS}${EXTRA_QUARKUS_EXTENSIONS}"

No one yet used something like that or needed that, but it is an option if
we want to add it and keep the nice defaults in the dockerfile. Anyhow, not
in this PR

**Review by gabriel-farache — APPROVED** [review]

**Review by masayag — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (rgolangh) [reply Makefile]: done

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -79,70 +76,36 @@ jobs:
...
-              }
-            }
-          }
-        )' ${{ inputs.workflow_id }}/manifests/01-sonataflow*.yaml
+        WORKDIR=${{ runner.temp }}/serverless-workflows
+        make WORKFLOW_ID=${{ inputs.workflow_id }} \
+          WORKDIR=${WORKDIR} \
+          APPLICATION_ID=${{ inputs.application_id }} \
+          PR_OR_COMMIT_URL=${PR_OR_COMMIT_URL} \
+          gen-manifests
```
[the remark this answers is not in the document]
Comment (rgolangh) [reply .github/workflows/main.yml:104]: handled in the top commit now

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -6,6 +6,11 @@ on:
       workflow_id:
         required: true
         type: string
+      # application is those auxilary application we deploy with a workflow to complete some part of it with specialized code, like for exaple the jira-listener for the escaplation flow.
```
[the remark this answers is not in the document]
Comment (rgolangh) [reply .github/workflows/main.yml]: fixed

**Review by masayag — APPROVED** [review]

**Comment by masayag** [comment]
@rgolangh https://github.com/parodos-dev/serverless-workflows/pull/317 was merged,
pls update this PR and let's see if that solves (some) of the issues.
The missing piece will probably be generating the token for the notifications.

**Comment by masayag** [comment]
@rgolangh the error is about incorrect request to invoke the MTA that now requires the recipient, which is an array that its elements should be in the format of `user:default/guest` or `user:development/guest`  (depending on the available namespaces).

so this line https://github.com/parodos-dev/serverless-workflows/blob/main/e2e/mta.sh#L31
```
out=$(curl -XPOST -H "Content-Type: application/json"  http://localhost:9080/api/orchestrator/workflows/MTAAnalysis/execute -d '{"repositoryURL": "https://github.com/spring-projects/spring-petclinic"}')
```
should be changed to:
```
out=$(curl -XPOST -H "Content-Type: application/json"  http://localhost:9080/api/orchestrator/workflows/MTAAnalysis/execute -d '{"repositoryURL": "https://github.com/spring-projects/spring-petclinic", "recipients": ["user:default/guest"]}')
```
that should be done for all workflows that requires notifications (all but greeting)

**Comment by rgolangh** [comment]
the state of this PR is that I'm working to get the upstream notification plugins working, because they don't at the moment.
We may need to think on how to mock/mimic/disable the notifications in some way, so e2e would pass,so we can merge this work because this PR will obviously get stale in time and we would loose the other

**Comment by gabriel-farache** [comment]
> We may need to think on how to mock/mimic/disable the notifications in some way
When the notification used to be optional, it was easy, we  simply had to not provide the recipients.

Now, we may need to startup a simple webserver (i.e: https://httpbin.org/) that does nothing and configure the notification URL to point to that dumb webserver

**Review by masayag — APPROVED** [review]

**Pull request merged**